### ISSUE TITLE
Possibly incorrect CA1067

### ISSUE THREAD

**Issue by onyxmaster** [dotnet/roslyn-analyzers#2027]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.6.3

#### Diagnostic ID

[CA1067](https://github.com/dotnet/roslyn-analyzers/blob/e878c81/src/Microsoft.CodeQuality.Analyzers/Documentation/CA1067_OverrideObjectEqualsWhenImplementingIEquatableOfT.md)

#### Repro steps

```
using System;

public class A<T> : IEquatable<T>
    where T : A<T>
{
    public virtual bool Equals(T other) => false;

    public override bool Equals(object obj) => Equals(obj as T);
}

public class B : A<B>
{
}
```

#### Expected behavior

No code analysis warnings are produced for class `B`.

#### Actual behavior

The definition of class `B` is annotated with warning CA1067 "Type B should override Equals because it implements IEquatable<T>".

#### Notes
If I recall correctly, v2.6.2 didn't exhibit this behavior.

**Comment by pantosha** [comment]
Any news?

In my opinition possible cause of this issue is fix of #1914.

**Comment by mavasani** [comment]
@pantosha That seems correct. Would you like to contribute a PR to fix it?

**Comment by onyxmaster** [comment]
Manish, thank you very much for this one, looking forward to new release.
